src/main/java/seedu/edrecord/storage/JsonAdaptedPerson.java

```diff
@@ -116,42 +116,45 @@ public Person toModelType() throws IllegalValueException {
-            }
-            final Module modelModule = Module.MODULE_SYSTEM.getModule(mod);
-
-            String groups = modGroupArray[1];
-            if (groups.length() <= 1 || groups.charAt(0) != '[' || groups.charAt(groups.length() - 1) != ']') {
-                throw new IllegalValueException(ModuleSet.MESSAGE_GROUP_CONSTRAINTS);
-            }
-            groups = ModuleSet.parseGroups(groups);
-            String[] groupArray = groups.split(",");
-
-            for (String group : groupArray) {
-                if (!Group.isValidGroup(group)) {
-                    throw new IllegalValueException(Group.MESSAGE_CONSTRAINTS);
+        if (!mods.equals("")) {
+            String[] modGroupPairs = mods.split(" ");
+            for (String modGroupPair : modGroupPairs) {
+                String[] modGroupArray = modGroupPair.split(":");
+                if (modGroupArray.length != 2) {
+                    throw new IllegalValueException(ModuleSet.MESSAGE_CONSTRAINTS);
+                }
+                String mod = modGroupArray[0];
+                if (!Module.isValidModuleCode(mod)) {
+                    throw new IllegalValueException(Module.MESSAGE_CONSTRAINTS);
                 }
-                if (!modelModule.getGroupSystem().hasGroup(group)) {
-                    throw new IllegalValueException(Group.MESSAGE_DOES_NOT_EXIST);
+                if (!Module.MODULE_SYSTEM.hasModule(mod)) {
+                    throw new IllegalValueException(String.format(Module.MESSAGE_DOES_NOT_EXIST, mod));
                 }
-                final Group modelGroup = modelModule.getGroup(group);
-                if (!modelModule.getGroupSystem().hasGroup(modelGroup)) {
-                    modelModule.getGroupSystem().addGroup(modelGroup);
+                final Module modelModule = Module.MODULE_SYSTEM.getModule(mod);
+
+                String groups = modGroupArray[1];
+                if (groups.length() <= 1 || groups.charAt(0) != '[' || groups.charAt(groups.length() - 1) != ']') {
+                    throw new IllegalValueException(ModuleSet.MESSAGE_GROUP_CONSTRAINTS);
+                }
+                groups = ModuleSet.parseGroups(groups);
+                String[] groupArray = groups.split(",");
+
+                for (String group : groupArray) {
+                    if (!Group.isValidGroup(group)) {
+                        throw new IllegalValueException(Group.MESSAGE_CONSTRAINTS);
+                    }
+                    if (!modelModule.getGroupSystem().hasGroup(group)) {
+                        throw new IllegalValueException(Group.MESSAGE_DOES_NOT_EXIST);
+                    }
+                    final Group modelGroup = modelModule.getGroup(group);
+                    if (!modelModule.getGroupSystem().hasGroup(modelGroup)) {
+                        modelModule.getGroupSystem().addGroup(modelGroup);
+                    }
+                    moduleSet.addToSet(modelModule);
```

Comment:
```suggestion
                    moduleSet.add(model, new Group(group));
```
Also, do not add the model system's module. The model module may have more groups than an individual person

Reply:
LGTM otherwise